examples/platforms/simulation/virtual_time/platform-sim.c

```diff
@@ -176,10 +177,26 @@ otError otPlatUartFlush(void)
 static void socket_init(void)
 {
     struct sockaddr_in sockaddr;
+    char *             base;
     char *             offset;
     memset(&sockaddr, 0, sizeof(sockaddr));
     sockaddr.sin_family = AF_INET;
 
+    base = getenv("PORT_BASE");
+
+    if (base)
+    {
+        char *endptr;
+
+        sPortBase = (uint16_t)strtol(base, &endptr, 0);
+
+        if (*endptr != '\0')
+        {
+            fprintf(stderr, "Invalid PORT_BASE: %s\n", base);
+            exit(EXIT_FAILURE);
+        }
+    }
+
```

Comment:
1. Make this a util function? 
2. Is this necessary for virtual time simulation?

Reply:
(Jumping in to comment, as I'm working also on virtual time simulation: )
Being able to change the base port from 9000 to something else, seems useful. There may be machines that don't have the ports 9000-9xxx free to do OT-NS simulations and this would provide a way to configure the base port.

On the naming, just wondering why the names are so generic (PORT_OFFSET as already present, and PORT_BASE).  Ideally they should have at least something like "OT_" prefixed to them to avoid unclarity of where the environment var belongs to. But ok to leave names as is.

Reply:
Agree with @EskoDijk. 
Yet, changing the name of `PORT_OFFSET` is well beyond the scope of this PR.

Feel free to submit PRs to refine `PORT_OFFSET`.

Reply:
@simonlingoogle Agree changing the name would be another PR, but not so important.

Just noting that still the value "9000" still appears hardcoded in various places in the codebase. Ideally there would be some constant defined in the right header file to denote the default UDP port for OT simulations / virtual-time features. That could also be a separate PR once this is integrated.